src/bsoncxx/lib/bsoncxx/v1/document/view.cpp

```diff
@@ -30,6 +43,167 @@ static_assert(
     is_nothrow_moveable<view::const_iterator>::value,
     "bsoncxx::v1::document::view::const_iterator must be nothrow moveable");
 
+namespace {
+
+constexpr std::uint8_t k_default_view[5] = {5u, 0u, 0u, 0u, 0u};
+
+} // namespace
+
+view::view() : view{k_default_view} {}
+
+view::view(std::uint8_t const* data, std::size_t length) : view{data} {
+    if (length < _empty_length || length < this->size()) {
+        throw v1::exception{v1::document::view::errc::invalid_length};
+    }
+}
+
+view::const_iterator view::cbegin() const {
+    if (!this->operator bool()) {
+        return this->cend();
+    }
+
+    bson_iter_t iter;
+
+    if (!bson_iter_init_from_data(&iter, _data, this->size())) {
+        throw v1::exception{code::invalid_data};
+    }
+
+    if (!bson_iter_next(&iter)) {
+        return this->cend();
+    }
+
+    return const_iterator::internal::make_const_iterator(
+        _data, this->size(), bson_iter_offset(&iter), bson_iter_key_len(&iter));
+}
+
+view::const_iterator view::find(v1::stdx::string_view key) const {
+    if (!this->operator bool()) {
+        return this->cend();
+    }
+
+    if (key.size() >= std::size_t{INT_MAX}) {
+        return this->cend();
+    }
```

Comment:
This check is [not present in v_noabi](https://github.com/mongodb/mongo-cxx-driver/blob/ad622035ec9c9ab352b42449d5fcbf426e8782e9/src/bsoncxx/lib/bsoncxx/v_noabi/bsoncxx/document/view.cpp#L111). This guards the cast to `int` in the later call to `bson_iter_init_find_w_len` (passing `-1` [does not help](https://github.com/mongodb/mongo-c-driver/blob/d3165e022ae9155289e1910c72c393d035f75658/src/libbson/src/bson/bson-iter.c#L304)).